rust/protocol/src/crypto.rs

```diff
@@ -61,45 +75,62 @@ pub fn aes_256_cbc_decrypt(ctext: &[u8], key: &[u8], iv: &[u8]) -> Result<Vec<u8
         .map_err(|_| SignalProtocolError::InvalidCiphertext)?)
 }
 
-pub fn hmac_sha256(key: &[u8], input: &[u8]) -> Result<[u8; 32]> {
+/// The statically-known size of the output of [hmac_sha256].
+pub const HMAC_OUTPUT_SIZE: usize = 32;
+
+/// Calculate the HMAC-SHA256 code over `input` using `key`.
+pub fn hmac_sha256(key: &[u8], input: &[u8]) -> [u8; HMAC_OUTPUT_SIZE] {
```

Comment:
That sounds great! I've made them all use `Result` for now. I do think it is very possible to perform that separate pass later and it will probably be useful at that point.